File: sdk-dotnet/InnovacionX.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;

public class InnovacionX
{
    private readonly string token;
    private readonly string baseUrl;
    private readonly HttpClient client;

    public InnovacionX(string token, string baseUrl = "https://tu-dominio.com/api/public")
    {
        if (string.IsNullOrEmpty(token))
            throw new Exception("Falta x-public-token");

        this.token = token;
        this.baseUrl = baseUrl;
        this.client = new HttpClient();
    }

    private async Task<string> Request(string endpoint)
    {
        var request = new HttpRequestMessage(HttpMethod.Get, baseUrl + endpoint);
        request.Headers.Add("x-public-token", token);
        request.Headers.Add("Content-Type", "application/json");

        var response = await client.SendAsync(request);

        if (!response.IsSuccessStatusCode)
            throw new Exception("HTTP Error " + response.StatusCode);

        return await response.Content.ReadAsStringAsync();
    }

    public Task<string> GetProductos()
    {
        return Request("/productos");
    }
}

File: sdk-java/InnovacionX.java
import java.net.http.*;
import java.net.URI;
import java.io.IOException;

public class InnovacionX {

    private String token;
    private String baseUrl;

    public InnovacionX(String token, String baseUrl) {
        if (token == null || token.isEmpty()) {
            throw new RuntimeException("Falta x-public-token");
        }
        this.token = token;
        this.baseUrl = baseUrl;
    }

    public InnovacionX(String token) {
        this(token, "https://tu-dominio.com/api/public");
    }

    private String request(String endpoint) throws IOException, InterruptedException {
        HttpClient client = HttpClient.newHttpClient();

        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(baseUrl + endpoint))
            .header("Content-Type", "application/json")
            .header("x-public-token", token)
            .GET()
            .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

        if (response.statusCode() >= 400) {
            throw new RuntimeException("Error HTTP " + response.statusCode());
        }

        return response.body();
    }

    public String getProductos() throws IOException, InterruptedException {
        return request("/productos");
    }
}
